NSFS | TypeError [ERR_INVALID_CHAR]: Invalid character in header content ["x-amz-meta-security.selinux"]

### Environment info
- NooBaa Version: **latest**
- Platform: **standalone**

### Actual behavior
1. Trying to head an object/file with security.selinux xattr fails:

```
[ERROR] core.endpoint.s3.s3_rest:: S3 ERROR <?xml version="1.0" encoding="UTF-8"?><Error><Code>InternalError</Code><Message>We encountered an internal error. Please try again.</Message>...

TypeError [ERR_INVALID_CHAR]: Invalid character in header content ["x-amz-meta-security.selinux"]
    at ServerResponse.setHeader (node:_http_outgoing:606:3)
    at set_response_xattr (/snapshot/noobaa/src/endpoint/s3/s3_utils.js:172:13)
    at Object.set_response_object_md (/snapshot/noobaa/src/endpoint/s3/s3_utils.js:331:5)
    at Object.head_object [as handler] (/snapshot/noobaa/src/endpoint/s3/ops/s3_head_object.js:29:14)
    at async handle_request (/snapshot/noobaa/src/endpoint/s3/s3_rest.js:148:19)
    at async Object.s3_rest [as handler] (/snapshot/noobaa/src/endpoint/s3/s3_rest.js:64:9)
```

### Expected behavior
1. Head object should work with security.selinux xattrs and any other attrs with dot

### Steps to reproduce
1. NA

### More information - Screenshots / Logs / Other output

--- a/src/endpoint/s3/s3_utils.js
+++ b/src/endpoint/s3/s3_utils.js
@@ -138,13 +138,17 @@ function format_s3_xml_date(input) {
 }
 
 const X_AMZ_META = 'x-amz-meta-';
+const HTTP_NB_DOT_CHAR = '__dot__';
 
 function get_request_xattr(req) {
     const xattr = {};
     _.each(req.headers, (val, hdr) => {
         if (!hdr.startsWith(X_AMZ_META)) return;
-        const key = hdr.slice(X_AMZ_META.length);
+        let key = hdr.slice(X_AMZ_META.length);
         if (!key) return;
+
+        // we encode unacceptable characters in the header name, reverting that here
+        key = key.replaceAll(HTTP_NB_DOT_CHAR, '.');
         xattr[key] = val;
     });
     return xattr;
@@ -162,7 +166,10 @@ function set_response_xattr(res, xattr) {
         keys.sort();
     }
     let returned_keys = 0;
-    for (const key of keys) {
+    for (let key of keys) {
+        // replace characters not allowed by RFC 7230 Section 3.2 with custom strings
+        key = key.replaceAll('.', HTTP_NB_DOT_CHAR);
+
         const md_header_size =
             X_AMZ_META.length +
             4 + // for ': ' and '\r\n'

